.github/workflows/ci-linux.yml

```diff
@@ -101,6 +102,7 @@ jobs:
       tox_packages_factors: >-
           ["standard"]
       docker_push_repository: ghcr.io/${{ github.repository }}/
+      max_parallel: 12
 
```

Comment:
I guess you picked up these numbers heuristically.  But what is the reason that 12 for "standard" is less than 15 for "standard-pre"?

Reply:
The heuristic was that "standard-pre" is rather important as you want to be notified early that there are profound compilation issues on a certain system. "standard" is a bit less important (and results seem to be more uniform across the systems tested), also there are a few systems dropping out as their pre step failed.

Reply:
Also note that there is an overall walltime limit of 3 days (?) for the whole workflow to pass. So whatever job restrictions we put in, we need to make sure that this does not cause the later workflows to never run.

Reply:
There is nothing crucial about these workflow runs. If they fail, then its just an ordinary bug that needs to be fixed. There is no need to have them finished as soon as possible. You can always remove some older/similar systems from the ci if things are then taking too long.

In my opinion it's not acceptable that other workflows are queued for 14+ hours because of it.

Reply:
> There is nothing crucial about these workflow runs. If they fail, then its just an ordinary bug that needs to be fixed. There is no need to have them finished as soon as possible.

Not sure for what audience you're writing this, Tobias...

Reply:
> Also note that there is an overall walltime limit of 3 days (?) for the whole workflow to pass. So whatever job restrictions we put in, we need to make sure that this does not cause the later workflows to never run.

That is 35 days according to https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration. So no problem here.

Reply:
Thanks for checking; then the time to the release (~ 7 days) is the closer deadline

Reply:
Yes.

Another question. What is the limit on the number of concurrent jobs? The linked page says that it depends on our plan. If we are on "Free" plan, it is 20 concurrent jobs. Is this the case?

Reply:
If it is 20, then even though we increase "max-parallel" to 50, only 20 of them will be run together in parallel. Am I right?

Reply:
The GitHub people gave us the Team plan for free, so we have 60 concurrent Linux runners (and 5 macOS runners).

Reply:
We have 45 systems in ci-linux. So even though we increase "max-parallel" to 50 for "standard-pre", "default" may not run before two passes of bundle of 20 parallel runs, unless "default" is lucky.

Reply:
Ah. 60 runners. Then increasing "max-parallel" to 50 is a good solution!

Also we may decrease "max-parallel" for "minimal", "maximal", "experimental" to allow rooms for other PR workflow runs.

Reply:
Sure. When I added the "standard-sitepackages" run (with a limit of 10 and running parallel with "standard"), I didn't decrease the max-parallel of "standard". We can reduce it a bit, from 30 perhaps to 25. I wouldn't go much lower than that.

Reply:
How about basing your PR on Tobias's?

Reply:
I've instead pushed a small change to https://github.com/sagemath/sage/pull/36616#issuecomment-1789410410

Reply:
> Thanks for checking; then the time to the release (~ 7 days) is the closer deadline

According to https://github.com/sagemath/sage/actions/runs/6598197147/usage, the overall run time is 28d, so an average number of 5 runners should be more than enough to make that deadline.